### PR TITLE
Feature catalog upgrade es5

### PR DESCRIPTION
**Pull request by EvanSnapp**

### CODE REVIEW

File: resource/environment/integration.js [hunk 1]
```diff
@@ -6,6 +6,11 @@ var format = require( "util" ).format;
 var environment = require( "../../src/data/nedb/environment" );
 var util = require( "../../src/util" );
 var dockerhub = require( "../../src/dockerhub" );
+var rp = require( "request-promise" );
+var yaml = require( "js-yaml" );
+function isnum ( val ) {
```
Comment (shawnHartsell) [inline resource/environment/integration.js]: use the lodash equivalant

File: resource/environment/integration.js [hunk 2]
```diff
@@ -241,6 +245,141 @@ function upgrade( slack, envelope ) {
 	} );
 }
 
+function getTemplate( token, catalogOwner, catalog, info ) {
```
Comment (shawnHartsell) [inline resource/environment/integration.js]: there are a lot of nested functions here. maybe consider putting this in a module?

File: resource/environment/integration.js [hunk 3]
```diff
@@ -241,6 +245,141 @@ function upgrade( slack, envelope ) {
...
+				access_token: token
+			},
+			headers: {
+				"User-Agent": "cowpoke"
+			},
+			json: true
+		} ).then( onResponse ).catch( console.log );
+}
+
+function upgradeStack( slack, github, envelope ) {
```
Comment (shawnHartsell) [inline resource/environment/integration.js]: it's a little difficult to reason about the code b/c of the amount of nested functions. may want to consider modularize this into modules

File: src/index.js [hunk 4]
```diff
@@ -8,15 +8,17 @@ var config = require( "configya" )( {
 } );
 var fount = require( "fount" );
 
-if ( !process.env.DOCKER_USER || !process.env.DOCKER_PASS ) {
+if ( !config.docker.user || !config.docker.pass ) {
```
Comment (shawnHartsell) [inline src/index.js:11]: should there be an entry for this in `config.json`?

**Comment by shawnHartsell** [comment]
@EvanSnapp thanks again for your work. Unfortunately the structure of the code makes it hard to understand it's intent

File: src/index.js (hunk 4 above)
Comment (EvanSnapp) [inline src/index.js]: I added it

Comment (EvanSnapp) [inline src/index.js:11]: I have now added it to the file in the amending commit

**Comment by shawnHartsell** [comment]
code is in the ES6 branch now :)

**Pull request closed without merging**